src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/src/components/api-authorization/AuthorizeService.js

```diff
@@ -0,0 +1,323 @@
+﻿import { UserManager, WebStorageStateStore } from 'oidc-client';
+import { ApplicationPaths, ApplicationName } from './ApiAuthorizationConstants';
+
+export class AuthorizeService {
+    _callbacks = [];
+    _nextSubscriptionId = 0;
+    _user = null;
+    _isAuthenticated = false;
+
+    async isAuthenticated() {
+        const user = await this.getUser();
+        return !!user;
+    }
+
+    async getUser() {
+        if (this._user && this._user.profile) {
+            return this._user.profile;
+        }
+
+        await this.ensureUserManagerInitialized();
+        const user = await this.userManager.getUser();
+        return user && user.profile;
+    }
+
+    async getAccessToken() {
+        await this.ensureUserManagerInitialized();
+        const user = await this.userManager.getUser();
+        return user && user.access_token;
+    }
+
+    // We try to authenticate the user in three different ways:
+    // 1) We try to see if we can authenticate the user silently. This happens
+    //    when the user is already logged in on the IdP and is done using a hidden iframe
+    //    on the client.
+    // 2) We try to authenticate the user using a PopUp Window. This might fail if there is a
+    //    Pop-Up blocker or the user has disabled PopUps.
+    // 3) If the two methods above fail, we redirect the browser to the IdP to perform a traditional
+    //    redirect flow.
+    async signIn(state) {
+        await this.ensureUserManagerInitialized();
+        try {
+            const silentUser = await this.userManager.signinSilent(this.createArguments(LoginMode.Silent));
+            this.updateState(silentUser);
+            return this.success(state);
+        } catch (silentError) {
+            // User might not be authenticated, fallback to popup authentication
+            console.log("Silent authentication error: ", silentError);
```

Comment:
Nit: Inconsistent quotes in a few places in this file

Reply:
Done